src/Server/WebSocketServer.php

```diff
@@ -162,10 +168,49 @@ private function onNewConnection(ConnectionInterface $socketStream)
             return $this->getMessageHandler($uri, $connection);
         }, $this->loop, $this->messageProcessor);
 
+        $socketStream->on('end', function () use($connection) {
+            $this->onDisconnect($connection);
+        });
+
         $connection->setLogger($this->getLogger());
+        $connection->getLogger()->info(sprintf('Ip "%s" establish connection', $connection->getIp()));
         $this->connections[] = $connection;
     }
 
+    /**
+     *
+     * @param Connection $connection
+     */
+    private function onDisconnect(Connection $connection)
+    {
+        $this->removeConnection($connection);
+        $connection->getLogger()->info(sprintf('Ip "%s" left connection', $connection->getIp()));
+
+        unset($connection);
+    }
+
+    /**
+     * Remove a Connection instance by his object id
+     * @param Connection $connection
+     * @param bool $strict Define if the process should stop if unfindable
+     * @throws RuntimeException This method throw an exception if the $connection instance object isn't findable in websocket server's connections
+     */
+    private function removeConnection(Connection $connection, bool $strict = true)
+    {
+        $connectionId = spl_object_hash($connection);
+        foreach ($this->connections as $index => $connectionIt) {
+            if ($connectionId === spl_object_hash($connectionIt)) {
+                unset($this->connections[$index]);
+                return;
+            }
+        }
+
+        $this->logger->critical(sprintf('Connection instance with id "%s" is unfindable in websocket server\'s active connections !', $connectionId));
```

Comment:
`Connection instance with id "%s" is unfindable in websocket server\'s active connections !` => `No connection found in the server connection list, impossible to delete the given connection id. Something wrong happened.`

(We don't care about the "id", an spl hash is not a useful information for debugging)